libr/core/carg.c

```diff
@@ -0,0 +1,248 @@
+        	r_core_read_at (core, bval, buf, bsize);
+	}
+	if (onstack) { // Fetch value from stack
+		bval = get_buf_val (buf, endian, width);
+		if (opt != 'd' && opt != 'x') {
+			r_core_read_at (core, bval, buf, bsize); // update buf with val from stack
+		}
+	}
+	switch (opt) {
+	case 'z' : // Null terminated string
+		r_cons_print ("\"");
+		for (i = 0; i < MAXSTRLEN; i++) {
+			if (buf[i] == '\0') {
+				break;
+			}
+			ut8 b = buf[i];
+			if (IS_PRINTABLE (b)) {
+				r_cons_printf ("%c", b);
+			} else {
+				r_cons_printf ("\\x%02x", b);
+			}
+			if (i == MAXSTRLEN - 1) {
+				 r_cons_print ("..."); // To show string is truncated
+			}
+		}
+		r_cons_print ("\"");
+		r_cons_newline ();
+		break;
+	case 'd' : // integer
+	case 'x' :
+		r_cons_printf ("0x%08" PFMT64x, bval);
+		r_cons_newline ();
+		break;
+	case 'c' : // char
+		r_cons_print ("\'");
+		ut8 ch = buf[0];
+		if (IS_PRINTABLE (ch)) {
+			r_cons_printf ("%c", ch);
+		} else {
+			r_cons_printf ("\\x%02x", ch);
+		}
+		r_cons_print ("\'");
+		r_cons_newline ();
+		break;
+	case 'p' : // pointer
+		{
+		// Try to deref the pointer once again
+		r_cons_printf ("0x%08"PFMT64x, get_buf_val (buf, endian, width));
+		r_cons_newline ();
+		break;
+		}
+	}
+	free (buf);
+}
+
+/* This functon display list of arg with some colors */
+
+R_API void r_core_print_func_args(RCore *core) {
+	RListIter *iter;
+	bool color = r_config_get_i (core->config, "scr.color");
+	if (!core->anal) {
+		return;
+	}
+	if (!core->anal->reg) {
+		return;
+	}
+	const char *pc = r_reg_get_name (core->anal->reg, R_REG_NAME_PC);
+	ut64 cur_addr = r_reg_getv (core->anal->reg, pc);
+	RAnalOp *op = r_core_anal_op (core, cur_addr);
+	if (!op) {
+		return;
+	}
+	if (op->type == R_ANAL_OP_TYPE_CALL) {
+		RAnalFunction *fcn;
+		RAnalFuncArg *arg;
+		int i;
+		int nargs = 0;
+		bool onstack = false;
+		const char *fcn_name = NULL;
+		ut64 pcv = op->jump;
+		if (pcv == UT64_MAX) {
+			pcv = op->ptr;
+		}
+		fcn = r_anal_get_fcn_at (core->anal, pcv, 0);
+		if (fcn) {
+			fcn_name = fcn->name;
+		} else {
+			if (core->flags) {
+				RFlagItem *item = r_flag_get_i (core->flags, pcv);
+				if (item) {
+					fcn_name = item->name;
+				}
+			}
+		}
+		RList *list = r_core_get_func_args (core, fcn_name);
+		if (color) {
+			r_cons_println (Color_BBLUE"------------------------------------"Color_RESET);
+		} else {
+			r_cons_println ("------------------------------------");
+		}
+		if (!r_list_empty (list)) {
+			int argcnt = 0;
+			r_list_foreach (list, iter, arg) {
+				if (arg->cc_source && !strncmp (arg->cc_source, "stack", 5)) {
+					onstack = true;
+				}
+				r_cons_printf (" arg [%d] - %s : ", argcnt, arg->name);
+				print_format_values (core, arg->fmt, onstack, arg->src);
+				argcnt++;
+			}
+		} else {
+			if (fcn) {
+				nargs = fcn->nargs;
+			}
+			if (nargs > 0) {
+				for (i = 0; i < nargs; i++) {
+					ut64 v = r_debug_arg_get (core->dbg, R_ANAL_CC_TYPE_STDCALL, i);
```

Comment:
No , this is done as worst-case scenario , see the func r_core_get_func_args() it fetches the arg depending on arch and function calling convention this is done only if that function fails  (also note these functions were taken out from disas.c and i felt that it's better to display something instead of void)

Reply:
yep